src/System.Management.Automation/engine/Interop/Windows/QueryDosDevice.cs

```diff
@@ -88,7 +88,8 @@ internal static string GetDosDeviceForNetworkPath(char deviceName)
                         }
                         else
                         {
-                            Diagnostics.Assert(false, "Really it is a dead code since GetDosDevice() is called only if PSDrive.DriveType == DriveType.Network");
+                            // TODO: This actually hit in a Parallels VM, so it's not quite dead yet!
+                            // Diagnostics.Assert(false, "Really it is a dead code since GetDosDevice() is called only if PSDrive.DriveType == DriveType.Network");
```

Comment:
Please open an issue about this and mark it "7.4-Blocking".

Reply:
Which commit, the debugger being broken or the executable failing on my VM, or both?

Reply:
Open an issue about the fact that this assertion is wrong on Parallels VM. This code path needs to be reviewed as the assumption is incorrect. Can you please find out what the `PSDrive.DriveType` is when running in Parallel VM?

Reply:
@andyleejordan Can you please check what `PSDrive.DriveType` is when the assertion fails in Parallel VM and share that info in #20043? Thanks!